src/SpeedyTransforms/legendre_ka.jl

```diff
@@ -0,0 +1,204 @@
+import SpeedyWeather.LowerTriangularMatrices: ij2k, get_lm_range, get_2lm_range
+import Atomix 
+
+using KernelAbstractions
+
+# (inverse) legendre transform kernel, called from _legendre!
+@kernel function inverse_legendre_kernel!(
+    g_north,                        # Scratch storage for legendre coefficients
+    g_south,                        # before fft
+    specs_data,                     # Data passed from spectral grid
+    legendre_polynomials_data,      # Pre-calculated Legendre coefficients
+    lmax,                           # Max l-value, from SpectralTransform struct
+    lon_offsets,                    # Longitude 
+    kjm_indices                     # precomputed indices for thread    
+)
+    tid = @index(Global)
+
+    if tid <= size(kjm_indices, 1)
```

Comment:
I think this isn't necassary with our new kernel launching logic, as it won't be launched on more threads than needed anyway.